watch/watch.go

```diff
@@ -82,14 +82,18 @@ func (w *watchable) Watch() (interface{}, Watch, error) {
 	}
 
 	c := make(chan struct{}, 1)
-	if w.value != nil {
-		c <- struct{}{}
-	}
-
+	notify := w.value != nil
 	w.active = append(w.active, c)
-	closeFn := w.closeFunc(c)
 	w.Unlock()
 
+	if notify {
+		select {
```

Comment:
this select is unnecessary? sending on this channel should be non-blocking since it's just created?